### PR TITLE
Stories/more public dashboards

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: db/warehouse/migrate/20210422191627_create_federal_census_breakdowns.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreateFederalCensusBreakdowns < ActiveRecord::Migration[5.2]
+  def change
+    create_table :federal_census_breakdowns do |t|
+      t.date :accurate_on, comment: 'Most recent census date'
+      t.string :type
+      t.string :geography_level, comment: 'State, zip, CoC (or maybe 010, 040, 050)'
+      t.string :geography, comment: 'MA, 02101, MA-500'
+      t.string :group, comment: 'All, age range, gender, this represents the sub-poplation you want to pull'
+      t.string :measure, comment: 'Detail of race, age, etc. (Asian, 50-59...)'
+      t.integer :value, comment: 'count of population'
```
Comment (awisema) [inline db/warehouse/migrate/20210422191627_create_federal_census_breakdowns.rb:10]: Should the scope fields have `index: true`?

File: drivers/public_reports/app/controllers/public_reports/warehouse_reports/homeless_populations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+###
+
+module PublicReports::WarehouseReports
+  class HomelessPopulationsController < ApplicationController
+    include WarehouseReportAuthorization
+    include ArelHelper
+    include PublicReports::WarehouseReports::PublicReportsControllerConcern
+
+    def overall
+      params[:pp] = 'disabled' # disable rack-mini-profiler
```
Comment (awisema) [inline drivers/public_reports/app/controllers/public_reports/warehouse_reports/homeless_populations_controller.rb]: Move this up to a `before_action`? Maybe even in the concern?

File: drivers/public_reports/app/models/public_reports/homeless_population.rb [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+
+    private def total_for(scope, population)
+      count = if population == :adults_with_children
+        scope.select(:household_id).distinct.count
+      else
+        scope.select(:client_id).distinct.count
+      end
+
+      word = case population
+      when :veterans
```
Comment (awisema) [inline drivers/public_reports/app/models/public_reports/homeless_population.rb:263]: `pluralize` is documented to take the singular, so 'Veteran', Household', and 'Person'?

File: drivers/public_reports/app/models/public_reports/homeless_population.rb [hunk 4]
```diff
@@ -0,0 +1,576 @@
...
+            data: data.to_a,
+            title: _('Household Composition'),
+            total: total_for(with_service_in_quarter(report_scope, date, population), population),
+          }
+        end
+      end
+    end
+
+    private def client_population(enrollment, date)
+      # NOTE: We may eventually want child-only households
```
Comment (awisema) [inline drivers/public_reports/app/models/public_reports/homeless_population.rb]: Obsolete comment?

File: drivers/public_reports/app/models/public_reports/homeless_population.rb [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+    private def with_service_in_quarter(scope, date, population)
+      service_scope = if population == :housed
+        GrdaWarehouse::ServiceHistoryService.where(shs_t[:date].gt(she_t[:move_in_date]))
+      else
+        :current_scope
+      end
+      scope_for(population, date, scope).with_service_between(start_date: date, end_date: date.end_of_quarter, service_scope: service_scope)
+    end
+
+    # NOTE: this count is equivalent to OutflowReport.exits_to_ph
```
Comment (awisema) [inline drivers/public_reports/app/models/public_reports/homeless_population.rb:319]: Obsolete comment?

**Review by awisema — CHANGES_REQUESTED** [review]

File: drivers/public_reports/app/models/public_reports/homeless_population.rb (hunk 5 above)
Comment (eanders) [reply drivers/public_reports/app/models/public_reports/homeless_population.rb:319]: I don't think so, the following still uses OutflowReport.

**Pull request merged**